wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/button/CommandXboxController.java

```diff
@@ -0,0 +1,346 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.wpilibj2.command.button;
+
+import edu.wpi.first.hal.FRCNetComm.tResourceType;
+import edu.wpi.first.hal.HAL;
+import edu.wpi.first.wpilibj.GenericHID;
+import edu.wpi.first.wpilibj.XboxController;
+
+/**
+ * Provides JoystickButtons for binding commands to an XboxController's buttons. Additionally offers
+ * getters for retrieving axis values.
+ */
+public class CommandXboxController extends GenericHID {
+  // reuses the Button and Axis enums from the original XboxController
+
+  private JoystickButton m_leftBumper;
+  private JoystickButton m_rightBumper;
+  private JoystickButton m_leftStick;
+  private JoystickButton m_rightStick;
+  private JoystickButton m_aButton;
+  private JoystickButton m_bButton;
+  private JoystickButton m_xButton;
+  private JoystickButton m_yButton;
+  private JoystickButton m_backButton;
+  private JoystickButton m_startButton;
+  private POVButton m_uButton; // 0 degrees
+  private POVButton m_urButton; // 45 degrees
+  private POVButton m_rButton; // 90 degrees
+  private POVButton m_drButton; // 135 degrees
+  private POVButton m_dButton; // 180 degrees
+  private POVButton m_dlButton; // 225 degrees
+  private POVButton m_lButton; // 270 degrees
+  private POVButton m_ulButton; // 315 degrees
```

Comment:
Could these be made `public final` and be initialized in the constructor? It would make accessing them slightly easier.

Reply:
We specifically wanted to avoid initializing them if they weren't in use.